File: archives/C#/0530. Minimum Absolute Difference in BST.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int GetMinimumDifference(TreeNode root) {
        Stack<TreeNode> rootStack=new Stack<TreeNode>();
        TreeNode node=root;
        int preValue=-1;
        int rep=int.MaxValue;
        while(node!=null || rootStack.Count()!=0){
            while(node!=null){
                rootStack.Push(node);
                node=node.left;
            }
            if(rootStack.Count()!=0){
                TreeNode curNode=rootStack.Pop();
                if(preValue==-1){
                    preValue=curNode.val;
                }
                else{
                    rep=Math.Min(rep,curNode.val-preValue);
                    preValue=curNode.val;
                }
                node=curNode.right;
            }
        }
        return rep;
    }
}

File: archives/20190519python/0530. Minimum Absolute Difference in BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def getMinimumDifference(self, root: TreeNode) -> int:
        rep=[]
        self.tree_to_list(root,rep)
        
        result=rep[1]-rep[0]
        for k in range(len(rep)-1):
            result=min(result,rep[k+1]-rep[k])
        return result
    
    def tree_to_list(self,root,rep):
        if not root:
            return
        
        self.tree_to_list(root.left,rep)
        rep.append(root.val)
        self.tree_to_list(root.right,rep)
